src/python/WMCore/MicroService/MSOutput/MSOutput.py

```diff
@@ -318,10 +317,16 @@ def makeSubscriptions(self, workflow):
                 self.logger.info(msg)
                 continue
 
-            self.logger.info("Performing rucio rule creation for workflow: %s, dataset: %s",
-                             workflow['RequestName'], dMap['Dataset'])
-
             ruleAttrs.update({'copies': dMap['Copies']})
+            # RelVals have a different lifetime setting and it depends on the sample type
+            if workflow['IsRelVal']:
+                ruleLifeT = self.relvalPolicy.getLifetimeByDataset(dMap['Dataset'])
+                ruleAttrs["lifetime"] = ruleLifeT
```

Comment:
Second question, does lifetime can be 0, e.g. if `getLifetimeByDataset` will return zero?

Reply:
This is a good point!
In theory it could be, but it doesn't make sense to create a Rucio rule with lifetime=0. Let me add some extra code to validate it as well.